examples/threading/threading.c

```diff
@@ -10,15 +10,43 @@
 
 void* threadfunc(void* thread_param)
 {
+    int ret_status;
 
     // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
     // hint: use a cast like the one below to obtain thread arguments from your parameter
-    //struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+    struct thread_data* thread_func_args = (struct thread_data *)thread_param;
+    
+    ret_status = usleep(thread_func_args->stm_wait_to_obtain_ms*1000);
+    if(ret_status != 0)
+    {
+	perror("usleep1");
+    }
+
+    ret_status = pthread_mutex_lock((thread_func_args->stm_mutex));
+    if(ret_status != 0)
+    {
+	perror("mutex lock");
+    }
+
+    ret_status = usleep(thread_func_args->stm_wait_to_release_ms*1000);
+    if(ret_status != 0)
+    {
+	perror("usleep2");
+    }
+
+    pthread_mutex_unlock((thread_func_args->stm_mutex));
+    if(ret_status != 0)
+    {
+	perror("mutex unlock");
+    }
+   
+    thread_func_args->thread_complete_success = true;
```

Comment:
@Rishab-Shah did the thread complete successfully if each of the steps above failed?